deltachat-ffi/deltachat.h

```diff
@@ -3240,6 +3275,38 @@ int             dc_msg_is_setupmessage        (const dc_msg_t* msg);
 char*           dc_msg_get_setupcodebegin     (const dc_msg_t* msg);
 
 
+/**
+ * Get url of a videochat invitation.
+ *
+ * Videochat invitations are sent out using dc_send_videochat_invitation()
+ * and dc_msg_get_viewtype() returns #DC_MSG_VIDEOCHAT_INVITATION for such invitations.
+ *
+ * @param msg The message object.
+ * @return If the message contains a videochat invitation,
+ *     the url of the invitation is returned.
+ *     If the message is no videochat invitation, NULL is returned.
+ *     Must be released using dc_str_unref() when done.
+ */
+char* dc_msg_get_videochat_url (const dc_msg_t* msg);
+
+
+/**
+ * Check if the videochat can be handled internally.
+ * If "basic webrtc" as of https://github.com/cracker0dks/basicwebrtc is used to initiate the videochat,
+ * this is returned by dc_msg_is_basic_videochat().
+ * "basic webrtc" videochat may be processed natively by the app
+ * whereas for other urls just the browser is opened.
+ *
+ * The videochat-url can be retrieved using dc_msg_get_videochat_url().
+ * To check if a message is a videochat invitation at all, check the message type for #DC_MSG_VIDEOCHAT_INVITATION.
+ *
+ * @param msg The message object.
+ * @return 0=message is no videochat invitation or cannot be handled internally
+ *     1=message is a videochat invitation that can be handled internally or by a supported browser.
+ */
+int dc_msg_is_basic_videochat (const dc_msg_t* msg);
```

Comment:
confusing documentation, i don't quite understand what this function does from the documentation

Reply:
As I understand it, it returns 1 if videochat invitation points to basicwebrtc instance. If it isn't, desktop should open external browser instead of internal basicwebrtc interface.

Reply:
> 0=message is no videochat invitation or cannot be handled internally

so if it is not a videochat invitation or invalid it should be opened in the browser? :thinking:

Reply:
But when I'm thinking about it the ui could decide better whether it can be handled internally or not, couldn't it?

Reply:
we used the tristate when we did not use the viewtype, however, together with the viewtype, there are still three state: before calling dc_msg_is_basic_videochat() one should check the viewtype:

```
if (dc_msg_get_viewtype() == DC_MSG_VIDEOCHAT_INVITATION) {
    url = dc_msg_get_videochat_url();
    if (dc_msg_is_basic_videochat()) {
        // videochat invitation that we ship a client for
    } else {
        // use browser for videochat, just open the url
    }
} else {
    // not a videochat invitation
}
```

wrt core or ui: if the receiver can know that bit without checking the server before the user clicks "join", maybe that can be handled in the ui. but maybe we want to show different icons, wording for internally/externally at some point. the core offers more flexibility in shipping that bit in some headers.  
but let's keep this point open, we can easily remove this function if it turns out, it is not needed.

Reply:
i added an example to deltachat.h